gateway/shard/shard.go

```diff
@@ -0,0 +1,81 @@
+// Package shard provides sharding capabilities through Manager, found in every
+// session.Session.
+package shard
+
+import (
+	"errors"
+	"strings"
+)
+
+// GenerateShardIDs generates an array of ints of 0..(total-1).
```

Comment:
```suggestion
// GenerateShardIDs generates a slice of integers of [0, total).
```
Arrays mean a different thing in Go.